common/search.ts

```diff
@@ -46,20 +45,14 @@ const GET_AUTOCOMPLETE_RESULTS = gql`
 
 class WatchSearchAPIConnector {
   apolloClient: ApolloClient<InMemoryCache>;
-  plan: PlanType;
+  plan: PlanContextFragment;
 
   constructor({ apolloClient, plan }) {
     this.apolloClient = apolloClient;
     this.plan = plan;
   }
 
   async onSearch(opts, queryConfig) {
-    // Not implemented, stub here just to silence the warning about
-    // unimplemented method.
-    return { results: [] };
-  }
-
-  async onAutocomplete(opts, queryConfig) {
```

Comment:
@terotik This was a pain to debug but it turns out the issue was an `onSearch` and `onAutocomplete` race condition. `onSearch` would resolve after `onAutocomplete` when querying searches for a second time, therefor overriding the list of results with an empty array. Any idea why we also used `onAutocomplete` here? Just using `onSearch` seems to do the trick.

Reply:
I'm getting `Uncaught No onAutocomplete handler provided and no Connector provided. You must configure one or the other.` warning on console running this locally. Could that be the reason? Not sure if it's a problem if it works though...

Reply:
That might be a problem in production builds, I'll take another look

Reply:
@terotik Can you test this locally again? I disabled autocomplete [here](https://github.com/kausaltech/kausal-watch-ui/pull/272/commits/83ae1463aee0994dd386ad062d7c3cf28296d851)